scep/certificate.go

```diff
@@ -0,0 +1,21 @@
+package scep
+
+import (
+	"crypto/x509"
+
+	"github.com/pkg/errors"
+)
+
+// CertOptions options with which to create and store a cert object.
+type CertOptions struct {
+	Leaf          *x509.Certificate
+	Intermediates []*x509.Certificate
+}
+
+func newCert(db DB, ops CertOptions) error {
+	err := db.StoreCertificate(ops.Leaf)
```

Comment:
The `authority.Sign` method will store the certificate to, why do we want to store it again?

Reply:
Ah, I wasn't aware of that and (incorrectly) assumed I needed the same logic as for ACME. I've removed the database from the implementation.

In SCEP there exists a flow in which a certificate is requested but not immediately created and signed. In this case, the client should start polling for its certificate. Approving the certificate is performed as a separate process (i.e. manual authorization). This flow likely requires some state to be kept.

The current state of the integration of SCEP doesn't support this and I also think that for step-ca this workflow might not be that interesting to support. If it has to be supported, a method for interacting with step-ca should be created that can used to approve a SCEP request. This could be part of step-cli.